### PR TITLE
[CHE-169] Add Express Async Errors Dep

### PR DESCRIPTION
**Pull request by seantokuzo**
### Description

As part of the CHE-167 BE Refactor Error Handling story, this PR
- adds the express-async-errors package
- Implements it by importing in index.ts

This package wraps the entire express application so in the case of asynchronous methods (middlewares and controllers), instead of manually calling `next` passing in an error, we can just throw an error and it will automatically be caught and passed to the `next` function, invoking the global error handler.

[express-async-errors npm package](https://www.npmjs.com/package/express-async-errors)

**Because of the added dependency, a Docker image rebuild will be required once story branch is merged into dev**
**Future testing on this story branch will require this updated image _OR_ a local image build**

### Jira Task

[JIRA TICKET](https://codehammers.atlassian.net/browse/CHE-169?atlOrigin=eyJpIjoiZmI1MzI5YWFmZDFlNGFhMWJjZGNmOTg1MThhMTI0ZDIiLCJwIjoiaiJ9)

### Testing Instructions

N/A - testing unavailable until future tickets making use of this new functionality

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [ ] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [ ] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recomended style guide.

### CODE REVIEW

File: server/errors/customError.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  constructor(message: string) {
+    super(message);
+
+    /*
+    This step is bc we are inheriting from a built in JS class
+    Helps correctly set the prototype chain of our custom error classes
+    Only really necessary if we are targeting es5 when transpiling typescript
+    */
+    Object.setPrototypeOf(this, CustomError.prototype);
+  }
```
Comment (brok3turtl3) [inline server/errors/customError.ts:19]: Great job commenting this - It was helpful! :clap:

File: server/index.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import path from "path";
 import express, { Request, Response, Application, NextFunction } from "express";
+import "express-async-errors";
```
Comment (brok3turtl3) [inline server/index.ts]: More image updates :grimacing:

**Review by brok3turtl3 — COMMENTED** [review]
@seantokuzo I love this new approach to error handling. I think it will work really well as we continue to scale out and I am eager to see it put into use. Great work! 🎉

**Review by brok3turtl3 — APPROVED** [review]

**Comment by brok3turtl3** [comment]
@seantokuzo I think you can go ahead and delete the PR for CHE-168 as all that code has been merged into this branch by the looks of it.

File: server/index.ts (hunk 2 above)
Comment (seantokuzo) [reply server/index.ts]: lol someday i'll stop adding/moving dependencies... someday

**Pull request merged**